jena-shacl/src/main/java/org/apache/jena/shacl/engine/constraint/CardinalityConstraint.java

```diff
@@ -38,7 +38,15 @@ public void validateNodeShape(ValidationContext vCxt, Graph data, Shape shape, N
         // Node shape with cardinality. Can this usefully be checked for in the parser?
         throw new ShaclParseException("Cardinality constraint on a node shape");
     }
-    
+
+    public int getMinCount(){
```

Comment:
Not sure if this the best thing to do. The subclass for max and min already have accessors.

Being an abstract class, CardinalityConstraint can't be created directly.

"MaxCount" have a "getMinCount" function is a bit strange. If there were a general min and max at the same time constraint, then on the superclass would make more sense but there isn't.

Please remove theses and use overrides in the subclasses.

Reply:
ok